src/sources/soundsourceproxy.cpp

```diff
@@ -731,6 +737,7 @@ mixxx::AudioSourcePointer SoundSourceProxy::openAudioSource(
                 // before exporting metadata. In this case the caller (this class)
                 // is responsible for updating the stream info if needed.
                 if (!m_pTrack) {
+                    // TODO: It is surprising that openAudioSource returns a m_pSoundSource
```

Comment:
Why is it suprising?? `SoundSource` is an extension of both `AudioSource` and `MetadataSource`, basic OO design.

Reply:
This should work and fix the bug with the open file:

```
m_pAudioSource = m_pSoundSource;
return m_pAudioSource;
```

Reply:
I have already fixed the underlying issue in https://github.com/mixxxdj/mixxx/pull/4584 please have a look.